### PR TITLE
Convert Hide Alert Checks from localStorage to user settings

### PR DESCRIPTION
**Pull request by rottencandy**
**Story**: 
https://issues.redhat.com/browse/ODC-5106

**Description**: 
Use `useUserSettings` hook to store the local storage data into a config map.

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/console-shared/src/components/health-checks/HealthChecksAlert.tsx [hunk 1]
```diff
@@ -17,13 +17,15 @@ import {
   DaemonSetModel,
   StatefulSetModel,
 } from '@console/internal/models';
+import { useUserSettingsCompatibility } from '@console/shared/src/hooks/useUserSettingsCompatibility';
```
Comment (invincibleJai) [inline frontend/packages/console-shared/src/components/health-checks/HealthChecksAlert.tsx:20]: nit: In same package can do relative imports

File: frontend/packages/console-shared/src/components/health-checks/HealthChecksAlert.tsx [hunk 2]
```diff
@@ -17,13 +17,15 @@ import {
...
 import { STORAGE_PREFIX } from '../../constants';
 import './HealthChecksAlert.scss';
 
 type HealthChecksAlertProps = {
   resource: K8sResourceKind;
 };
 
+const HEALTH_CHECK_CONFIGMAP_KEY = 'console.healthChecks';
```
Comment (invincibleJai) [inline frontend/packages/console-shared/src/components/health-checks/HealthChecksAlert.tsx:28]: `console` is used across for CM key prefix , will it make sense to keep in in const ?

Comment (rottencandy) [reply frontend/packages/console-shared/src/components/health-checks/HealthChecksAlert.tsx]: Moved.

File: frontend/packages/console-shared/src/constants/common.ts [hunk 3]
```diff
@@ -30,6 +30,8 @@ export const UNASSIGNED_APPLICATIONS_KEY = '#UNASSIGNED_APP#';
 // Prefix our localStorage items to avoid conflicts if another app ever runs on the same domain.
 export const STORAGE_PREFIX = 'bridge';
 
+export const CONFIGMAP_PREFIX = 'console';
```
Comment (jerolimov) [inline frontend/packages/console-shared/src/constants/common.ts:33]: @rottencandy @invincibleJai Can / should we call this `USERSETTINGS_PREFIX` ? :smile:

File: frontend/packages/console-shared/src/hooks/__tests__/useUserSettings.spec.ts [hunk 4]
```diff
@@ -54,20 +55,22 @@ describe('useUserSettings', () => {
...
-      const [settings, setSettings] = useUserSettings('devconsole.topology.key', 'list');
-      expect(setSettings).toBeDefined();
-      waitAndExpect(() => {
-        expect(settings).toEqual('list');
-        expect(spyK8sPatch).toHaveBeenCalledTimes(1);
-        done();
-      });
+      [settings, callback] = useUserSettings('devconsole.topology.key', 'list');
+      expect(callback).toBeDefined();
+    });
```
Comment (jerolimov) [inline frontend/packages/console-shared/src/hooks/__tests__/useUserSettings.spec.ts:68]: :+1:

Comment (rottencandy) [reply frontend/packages/console-shared/src/hooks/__tests__/useUserSettings.spec.ts:68]: This is part of Sahil's PR

File: frontend/packages/console-shared/src/constants/common.ts (hunk 3 above)
Comment (rottencandy) [reply frontend/packages/console-shared/src/constants/common.ts]: Updated.

**Comment by invincibleJai** [comment]
/lgtm 

Verified the changes , works as expected

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**